bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,432 @@
+            String text = node.getEntityName();
+            String id = node.getId();
+            String type;
+            if (node instanceof Process) {
+                type = "node:function";
+            } else if (node instanceof Store) {
+                type = "node:storage";
+            } else if (node instanceof External) {
+                type = "node:input-output";
+            } else {
+                type = "error";
+            }
+
+            List<WebEditorLabel> labels = new ArrayList<>();
+            for (Label label : node.getProperties()) {
+                String labelTypeId = ((LabelType) label.eContainer()).getId();
+                String labelId = label.getId();
+                labels.add(new WebEditorLabel(labelTypeId, labelId));
+            }
+
+            List<Port> ports = new ArrayList<>();
+
+            for (Pin pin : node.getBehaviour().getInPin()) {
+                ports.add(new Port(null, pin.getId(), "port:dfd-input", new ArrayList<>()));
+            }
+
+            Map<Pin, List<AbstractAssignment>> mapPinToAssignments = mapping(node);
+
+            for (Pin pin : node.getBehaviour().getOutPin()) {
+                String behaviour = createBehaviourString(mapPinToAssignments.get(pin));
+                ports.add(new Port(behaviour, pin.getId(), "port:dfd-output", new ArrayList<>()));
+            }
+
+            children.add(new Child(text, labels, ports, id, type, null, null, new ArrayList<>()));
+        }
+    }
+
+    private void createFlows(DataFlowDiagram dataFlowDiagram, List<Child> children) {
+        for (Flow flow : dataFlowDiagram.getFlows()) {
+            String id = flow.getId();
+            String type = "edge:arrow";
+            String sourceId = flow.getSourcePin().getId();
+            String targetId = flow.getDestinationPin().getId();
+            String text = flow.getEntityName();
+            mapInputPinToFlowName.put(flow.getDestinationPin(), text);
+            children.add(new Child(text, null, null, id, type, sourceId, targetId, new ArrayList<>()));
+        }
+    }
+
+    private Map<Pin, List<AbstractAssignment>> mapping(Node node) {
+        Map<Pin, List<AbstractAssignment>> mapPinToAssignments = new HashMap<>();
+
+        for (AbstractAssignment assignment : node.getBehaviour().getAssignment()) {
+            if (mapPinToAssignments.containsKey(assignment.getOutputPin())) {
+                mapPinToAssignments.get(assignment.getOutputPin()).add(assignment);
+            } else {
+                List<AbstractAssignment> list = new ArrayList<>();
+                list.add(assignment);
+                mapPinToAssignments.put(assignment.getOutputPin(), list);
+            }
+        }
+        return mapPinToAssignments;
+    }
+
+    private String createBehaviourString(List<AbstractAssignment> abstractAssignments) {
+        StringBuilder builder = new StringBuilder();
+        if (abstractAssignments != null) {
```

Comment:
Replace this if with a guard (e.g. returning a result early when `abstractAssignments` is null)